lib/index.js

```diff
@@ -75,10 +78,35 @@ Mixpanel.prototype.identify = function(identify, fn){
     data: b64encode(payload)
   };
 
-  this
-    .get('/engage')
-    .query(query)
-    .end(this._parseResponse(fn));
+  // $set must be a separate call than $union
+  batch.push(function(fn){
+    self
+      .get('/engage')
+      .query(query)
+      .end(self._parseResponse(fn));
+  });
+
+  // https://mixpanel.com/help/reference/ios-push-notifications
+  // Send device token for iOS for push notification support
+  var deviceToken = identify.proxy('context.device.token');
+
+  if (deviceToken) {
+    delete payload.$set;
```

Comment:
This is pretty brittle; make a base `payload` and extend call-specific objects instead. E.g.:

```js
var basePayload = {
  // ...
};

batch.push(function(fn) {
  var payload = extend({
    $set: // ...
  }, basePayload);
  self
    .get('/engage')
    .query(b64encode(payload))
    .end(self._parseResponse(fn));
});
```